app/scripts/controllers/pod.js

```diff
@@ -118,6 +226,9 @@ angular.module('openshiftConsole')
               setLogVars(pod);
               setContainerVars();
             }));
+            
+            // TODO Proper location for this call?
+            populateFrontEndTerminalData($scope.pod.status.containerStatuses);
```

Comment:
You can probably call this function in two places.  Once above and once below the watch:

```javascript
// populate with data from DataService.get (this happens once)
populateFrontEndTerminalData(pod.status.containerStatuses);
watches.push(DataService.watchObject, ...
    // now, update with DataService.watch data (this happens at least once, potentially many times)
    populateFrontEndTerminalData(pod.status.containerStatuses);

```